astrodb_scripts/photometry.py

```diff
@@ -0,0 +1,405 @@
+    telescope: str
+        Telescope name
+    instrument: str
+        Instrument name
+    filter_name: str
+        Filter name
+
+    Returns
+    -------
+    filter_id: str
+        Filter ID
+    wave_eff: Quantity
+        Effective wavelength
+    fwhm: Quantity
+        Full width at half maximum (FWHM)
+    width_effective: Quantity
+        Effective width of the filter
+
+    Raises
+    ------
+    AstroDBError
+        If the SVO URL is not reachable or the filter information is not found
+    KeyError
+        If the filter information is not found in the VOTable
+    """
+
+    if internet_connection() is False:
+        msg = (
+            "No internet connection. "
+            "Cannot fetch photometry filter information from the SVO website."
+        )
+        logger.error(msg)
+        raise AstroDBError(msg)
+
+    url = (
+        f"http://svo2.cab.inta-csic.es/svo/theory/fps3/fps.php?ID="
+        f"{telescope}/{instrument}.{filter_name}"
+    )
+    r = requests.get(url)
+
+    if r.status_code != 200:
+        msg = f"Error retrieving {url}. Status code: {r.status_code}"
+        logger.error(msg)
+        raise AstroDBError(msg)
+
+    # Parse VOTable contents
+    content = BytesIO(r.content)
+    votable = parse(content)
+
+    # Get Filter ID
+    try:
+        filter_id = votable.get_field_by_id("filterID").value
+    except KeyError:
+        msg = f"Filter {telescope}, {instrument}, {filter_name} not found in SVO."
+        raise AstroDBError(msg)
+
+    # Get effective wavelength and FWHM
+    wave_eff = votable.get_field_by_id("WavelengthEff")
+    fwhm = votable.get_field_by_id("FWHM")
+    width_effective = votable.get_field_by_id("WidthEff")
+
+    if wave_eff.unit == "AA" and fwhm.unit == "AA" and width_effective.unit == "AA":
+        wave_eff = wave_eff.value * u.Angstrom
+        fwhm = fwhm.value * u.Angstrom
+        width_effective = width_effective.value * u.Angstrom
+    else:
+        msg = (
+            f"Wavelengths from SVO may not be Angstroms as expected: {wave_eff.unit},"
+            f"{fwhm.unit}, {width_effective.unit}."
+        )
+        raise AstroDBError(msg)
+
+    logger.debug(
+        f"Found in SVO: "
+        f"Filter {filter_id} has effective wavelength {wave_eff} and "
+        f"FWHM {fwhm} and effective width {width_effective}."
+    )
+
+    return filter_id, wave_eff, fwhm, width_effective
+
+
+def assign_ucd(wave_eff_quantity: u.Quantity):
+    """
+    Assign a Unified Content Descriptors (UCD) to a photometry filter
+    based on its effective wavelength
+    UCDs are from the UCD1+ controlled vocabulary
+    https://www.ivoa.net/documents/UCD1+/20200212/PEN-UCDlist-1.4-20200212.html#tth_sEcB
+
+    Parameters
+    ----------
+    wave_eff: Quantity
+        Effective wavelength
+
+    Returns
+    -------
+    ucd: str
+        UCD string
+
+    """
+    wave_eff_quantity.to(u.Angstrom)
+    wave_eff = wave_eff_quantity.value
+
+    if 3000 < wave_eff <= 4000:
```

Comment:
it'd probably be a bit more readable to have this in a dictionary

Reply:
template: https://stackoverflow.com/questions/53853276/mapping-range-of-values-to-string